purchase_discount/models/stock_move.py

```diff
@@ -7,6 +7,11 @@
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    def write(self, values):
+        if self.env.context.get("skip_update_price_unit"):
+            return True
```

Comment:
A mix of @HviorForgeFlow and @sergio-teruel suggestions:

```suggestion
        if self.env.context.get("skip_update_price_unit") and vals.get("price_unit"):
            vals.pop("price_unit")
```